tools/render-test/shader-input-layout.cpp

```diff
@@ -440,6 +440,23 @@ namespace renderer_test
         }
     }
 
+    template <typename F>
+    void _iteratePixels(int dimension, int size, unsigned int * buffer, F f)
+    {
+        if (dimension == 1)
```

Comment:
Not a problem with this PR, but the original code is an example of why I think the word "dimension" shouldn't be used as a name, especially for something like a bare `int`. It isn't clear until one inspects the code that `dimension` determines the rank/arity/dimensionality of the image, and not a single axis/dimension to iterate along.